lib/pbench/cli/agent/commands/tools/list.py

```diff
@@ -36,13 +36,17 @@ def execute(self):
             host_tools = {}
             for group in groups:
                 host_tools[group] = {}
-                for path in self.gen_tools_group_dir(group).glob("*/**"):
-                    if self.context.with_option:
-                        host_tools[group][path.name] = [
-                            p.read_text() for p in self.tools(path)
-                        ]
-                    else:
-                        host_tools[group][path.name] = [p for p in self.tools(path)]
+                try:
+                    for path in self.gen_tools_group_dir(group).glob("*/**"):
+                        if self.context.with_option:
+                            host_tools[group][path.name] = [
+                                (p, (path / p).read_text()) for p in self.tools(path)
+                            ]
+                        else:
+                            host_tools[group][path.name] = [p for p in self.tools(path)]
+                except BadToolGroup:
```

Comment:
If there's a `BadToolGroup` in the list of `groups`, should we be returning, or just reporting an error and continuing to the next iteration?

Reply:
I don't think it matters much in this case: we are either specifying a single group with `-g` or we don't specify anything and then it's all existing groups. In the first case, we return immediately, in the second we don't have any bad groups by definition. Right now, multiple `-g` options are allowed but are ineffective: the last one wins. But if we change that, then we might want to change the error behavior as well. 

However, this PR is just meant to bring `pbench-list-tools` to parity with the 0.69.9 one, so the changes are deliberately very minimal. In the longer run, we need a layer that reads the on-disk structure and constructs an in-memory structure that can answer whatever questions might arise from any of these utilities and I'm not convinced that we have that yet.